static/app/views/performance/queues/destinationSummary/messageSpanSamplesPanel.tsx

```diff
@@ -128,6 +137,13 @@ export function MessageSpanSamplesPanel() {
   sampleFilters.addFilterValue('transaction', query.transaction);
   sampleFilters.addFilterValue('messaging.destination.name', query.destination);
 
+  // filter by key-value filters specified in the search bar query
+  Object.entries(new MutableSearch(query.spanSearchQuery).filters).forEach(
+    ([key, value]) => {
+      sampleFilters.addFilterValues(key, value);
+    }
+  );
+
```

Comment:
😭 understandable if there's no way around this, but yuck! More motivation to improve `MutableSearch`

Reply:
yea.. I was looking at changing `MutableSearch.addStringFilter` to support strings containing multiple subfilters, but we would have to add some gnarly parsing logic that seems quite error-prone. I'll look into improving this after I add in the rest of the search bars.